### PR TITLE
Dev profile settings

### PR DESCRIPTION
**Pull request by oliverboamah**
Add Bloc to settings and change password screen.

### CODE REVIEW

File: lib/screens/profile/password_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  PasswordBloc({@required this.passwordRepository})
+      : assert(passwordRepository != null);
+
+  @override
+  PasswordState get initialState => PasswordInitialState();
+
+  @override
+  Stream<PasswordState> mapEventToState(PasswordEvent event) async* {
+    if (event is ChangePasswordEvent) {
+      if (event.currentPassword.length == 0) {
```
Comment (juanagu) [inline lib/screens/profile/password_bloc.dart]: could you replace for "currentPassword.isEmpty" the other validations too.

File: lib/repos/settings_repository.dart [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class SettingsRepository {
+  Future<void> updateFullName(String fullName) async {}
```
Comment (juanagu) [inline lib/repos/settings_repository.dart]: Maybe a good mock behavior could be Future.delayed(Duration(seconds: N,));

**Review by juanagu — CHANGES_REQUESTED** [review]

File: lib/repos/settings_repository.dart (hunk 2 above)
Comment (oliverboamah) [reply lib/repos/settings_repository.dart]: Okay cool.

File: lib/screens/profile/password_bloc.dart (hunk 1 above)
Comment (codingoliver) [reply lib/screens/profile/password_bloc.dart]: Please I don't understand what you mean. could you elaborate?

Comment (oliverboamah) [reply lib/screens/profile/password_bloc.dart]: Fixed it

**Review by umair13adil — APPROVED** [review]

**Comment by 4seer** [comment]
The biggest problem with the update is that the settings screen was moved into profile component 3 days ago and is part of profile bloc now. Therefore the updates should be applied to profile bloc and not to settings bloc because it is not separate anymore.

**Pull request merged**